### ISSUE TITLE
clockwise/counter clockwise does not seem to work reliably

### ISSUE THREAD

**Issue by marmotghost** [Qrrbrbirlbel/tikz-extensions#2]
Thanks for all the extensions! I believe I may have come across a possible issue. The clockwise/counter clockwise keys do not seem to give one what one may expect.
```
\documentclass[tikz,border=3mm]{standalone}
\usetikzlibrary{ext.paths.arcto}
\begin{document}
\begin{tikzpicture}[radius=1,-stealth]
 \scoped[red]{\draw (0,0)  arc to[clockwise]  (2,0);}
 \scoped[shift={(0,-3)}]{\draw (0,0) arc to[counter clockwise] (2,0);}
 \scoped[shift={(3,0)}]{\draw (0,0) arc to[clockwise] (1,1);}
 \scoped[shift={(3,-3)}]{\draw (0,0) arc to[counter clockwise] (1,1);}
\end{tikzpicture}
\end{document}
```
<img width="384" alt="Screen Shot 2022-10-01 at 10 04 29 AM" src="https://user-images.githubusercontent.com/68924170/193420316-73cfe140-46ed-402b-8c61-d946a078b969.png">
The unexpected result is marked red.

To be fair, this is not really caused by the library, but rather pgf. The command `\pgfpatharcto` accepts two flags, large/small and clockwise/counter clockwise, which may clash because one of them already uniquely determines the arc in all cases except when the start angle and end angle differ by 180. A possible way to resolve the issue in the library is to distinguish the cases within the library, precompute the data for `\pgfpatharctoprecomputed` and use the latter, which has the additional benefit to yield more accurate results.

**Comment by Qrrbrbirlbel** [comment]
Interesting! The macro `\pgfpatharcto` is missing a special condition which we can patch seemingly pretty easily:

```latex
\documentclass[tikz,border=3mm]{standalone}
\usetikzlibrary{ext.paths.arcto}
\makeatletter
\usepackage{etoolbox}
\patchcmd\pgfpatharcto{\ifdim\pgf@arcto@fA pt=0pt\ifdim}{%
  \ifdim\pgf@arcto@delta@theta pt=180pt
    \ifdim\pgf@arcto@fS pt=0pt
       \pgfmathsetmacro\pgf@arcto@theta@end{\pgf@arcto@theta@end+360}
    \fi
  \fi
  \ifdim\pgf@arcto@fA pt=0pt \ifdim}{}{}
\makeatother
\begin{document}
\begin{tikzpicture}[radius=1,-stealth]
 \scoped[red]{\draw (0,0)  arc to[clockwise]  (2,0);}
 \scoped[shift={(0,-3)}]{\draw (0,0) arc to[counter clockwise] (2,0);}
 \scoped[shift={(3,0)}]{\draw (0,0) arc to[clockwise] (1,1);}
 \scoped[shift={(3,-3)}]{\draw (0,0) arc to[counter clockwise] (1,1);}
\end{tikzpicture}
\end{document}
```

You're [not the first one to suggest using `\pgfpatharctoprecomputed`](https://texwelt.de/fragen/1490/wie-kann-ich-mit-tikz-einen-kreisbogen-zwischen-zwei-punkten-zeichnen/1503) (see comment by cfeuersaenger). Least of all PGF/TikZ itself doesn't recommend `\pgfpatharcto`.

Yes, the `\pgfpatharctoprecomputed` macro would be much better but then I have to do the math of [calculating an ellipse through two points](https://stackoverflow.com/a/198137/2163894) and even worse I'm losing the timer for nodes I'm smuggling out of `\pgfpatharcto` (though, we could use the calculations of `\pgfpatharcto` even if the result isn't accurate: for placing nodes it doesn't need to be so hyper-precise).

I would patch `\pgfpatharcto` for the next update, I don't know about `\pgfpatharctoprecomputed`, yet.

**Comment by Qrrbrbirlbel** [comment]
This wasn't really fixed with version 0.3.1/0.4 since, apparently I've somehow introduced two typos. With 0.4.1 it should work now.

Since the math is a bit wonky and it also failed for values close to 180° I've just added a check whether start and end angles are actually following the counter/clockwise flag:

https://github.com/Qrrbrbirlbel/tikz-extensions/blob/86667dfd029ac72f705b94e1baf510b3a8a4bf41/tex/generic/tikz-ext/tikzlibraryext.paths.arcto.code.tex#L305-L317

This also means that the library contains the full `\pgfpatharcto` code (and doesn't PGF's own anymore) because there wasn't a good hook for this patch.

@marmotghost Actually, your red arc is correct and the black counter-clockwise is wrong but that's all the same.
